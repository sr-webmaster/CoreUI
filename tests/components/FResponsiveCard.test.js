import { mount } from '@vue/test-utils'
import Component from '@freshinup/core-ui/src/components/FResponsiveCard.vue'

describe('FResponsiveCard', () => {
  it('snapshot desktop', async () => {
    const wrapper = mount(Component, {
      propsData: {
        cardClass: 'test-class'
      },
      mocks: {
        $vuetify: { breakpoint: { xsOnly: false } }
      },
      slots: {
        default: '<div class="slot" />'
      }
    })

    expect(wrapper.find('.v-card').classes()).toContain('test-class')
    expect(wrapper.find('.slot').exists()).toBe(true)
    expect(wrapper.find('.v-expansion-panel').exists()).toBe(false)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('snapshot mobile', async () => {
    const wrapper = mount(Component, {
      propsData: {
        cardClass: 'test-class',
        title: 'Test Title'
      },
      mocks: {
        $vuetify: { breakpoint: { xsOnly: true } }
      },
      slots: {
        default: '<div class="slot" />'
      }
    })

    expect(wrapper.find('.v-expansion-panel__header').text()).toContain('Test Title')
    expect(wrapper.find('.v-card').classes()).toContain('test-class')
    expect(wrapper.find('.slot').exists()).toBe(true)
    expect(wrapper.find('.v-expansion-panel').exists()).toBe(true)
    expect(wrapper.element).toMatchSnapshot()
  })
})
