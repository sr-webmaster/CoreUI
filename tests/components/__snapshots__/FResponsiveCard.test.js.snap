// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FResponsiveCard snapshot desktop 1`] = `
<div
  class="v-card v-sheet theme--light test-class"
>
  <div
    class="slot"
  />
</div>
`;

exports[`FResponsiveCard snapshot mobile 1`] = `
<ul
  class="v-expansion-panel v-expansion-panel--focusable theme--light"
>
  <li
    class="v-expansion-panel__container"
    tabindex="0"
  >
    <div
      class="v-expansion-panel__header"
    >
      <h3>
        Test Title
      </h3>
      <div
        class="v-expansion-panel__header__icon"
      >
        <i
          aria-hidden="true"
          class="v-icon material-icons theme--light"
        >
          keyboard_arrow_down
        </i>
      </div>
    </div>
    <div
      class="v-expansion-panel__body"
      style="display: none;"
    >
       
      <hr
        class="v-divider theme--light"
      />
       
      <div
        class="v-card v-sheet theme--light test-class"
      >
        <div
          class="slot"
        />
      </div>
    </div>
  </li>
</ul>
`;
